controllers/dopplersecret_controller_secrets.go

```diff
@@ -216,20 +256,37 @@ func (r *DopplerSecretReconciler) UpdateSecret(ctx context.Context, dopplerSecre
 	if versErr != nil {
 		return fmt.Errorf("Failed to compute processors version: %w", versErr)
 	}
+
+	currentNameTransformerVersion, versErr := GetNameTransformersVersion(dopplerSecret.Spec.NameTransformer)
+	if versErr != nil {
+		return fmt.Errorf("Failed to compute name transform version: %w", versErr)
+	}
+
 	log.Info("Fetching Doppler secrets")
 	secretVersion := ""
+
+	// Secret processors
 	processorsVersion := ""
+	nameTransformerVersion := ""
 	if existingKubeSecret != nil {
 		secretVersion = existingKubeSecret.Annotations[kubeSecretVersionAnnotation]
 		processorsVersion = existingKubeSecret.Annotations[kubeSecretProcessorsVersionAnnotation]
+		nameTransformerVersion = existingKubeSecret.Annotations[kubeSecretNameTransformerVersionAnnotation]
 	}
 
 	processorsVersionChanged := currentProcessorsVersion != processorsVersion
+	nameTransformerVersionChanged := currentNameTransformerVersion != nameTransformerVersion
 	requestedSecretVersion := secretVersion
 
-	if processorsVersionChanged {
+	// If processors or name transformer have changed, requestedSecretVersion to an empty secret version to reload the secrets
+	if processorsVersionChanged && !nameTransformerVersionChanged {
```

Comment:
Good thinking on including this info in the logs! However, I think it might be better if we keep this as one if block like:

```go
processorsVersionChanged := currentProcessorsVersion != processorsVersion
nameTransformerVersionChanged := currentNameTransformerVersion != nameTransformerVersion
requestedSecretVersion := secretVersion

if processorsVersionChanged || nameTransformerVersionChanged {
    log.Info("[/] Secret modifiers changed, reloading secrets.", "nameTransformersChanged", nameTransformerVersionChanged, "processorsChanged", processorsVersionChanged)
    requestedSecretVersion = ""
}
```

I feel like this approach is a bit less error-prone, especially if we plan to add more of these. What do you think?

Reply:
Good call~